python/GafferUI/Widget.py

```diff
@@ -1206,7 +1206,8 @@ def __startDrag( self, qObject, qEvent ) :
 			self.__lastButtonPressWidget = None
 			return False
 
-		if ( self.__lastButtonPressEvent.line.p0 - self.__positionToLine( qEvent.pos() ).p0 ).length() < 3 :
+		widget = Widget._owner( qObject )
```

Comment:
I think this was probably subtly wrong before, but it's more obvious with the call to `Widget._owner()`. `self.__lastButtonPressEvent` is in the coordinate space for `self.__lastButtonPressWidget`, but `Widget._owner( qObject )` is not necessarily the same as `self.__lastButtonPressEvent` so we can end up comparing coordinates in different spaces. I think we need to drop the call to `_owner()` and move this distance check down so we can use `__widgetSpaceLine( qEvent, sourceWidget )` in the comparison instead.

Reply:
I may be missing something, but does [this](https://github.com/GafferHQ/gaffer/pull/4053/commits/223c5297f7d24a51ff0333fcabf53bcf295bfed8) work too?

Reply:
`__lastButtonPressWidget()` evaluates a weak reference, so may return `None`. We should use `__widgetSpaceLine( qEvent, sourceWidget )`, only after we have checked that `sourceWidget` isn't `None`.

Reply:
Sorry, brain fail, had missed that weren't checking the `weakref` at the top. Fixed [here](https://github.com/GafferHQ/gaffer/commit/6fa5535d2ffdac808218e6a1b6871b2176b109e7).

Reply:
Thanks, could you squash down for merging please?

Reply:
Done